forte/data/data_store.py

```diff
@@ -0,0 +1,261 @@
+            #               "chunk", "ner", "sense", "is_root", "ud_features",
+            #               "ud_misc"],
+            #     "ft.onto.base_ontology.Document": ["document_class",
+            #               "sentiment", "classifications"],
+            #     "ft.onto.base_ontology.Sentence": ["speaker", "part_id",
+            #               "sentiment", "classification","classifications"],
+            # }
+            # self._type_attributes is:
+            # {
+            #     "ft.onto.base_ontology.Token": {"pos": 4, "ud_xpos": 5,
+            #               "lemma": 6, "chunk": 7, "ner": 8, "sense": 9,
+            #               "is_root": 10, "ud_features": 11, "ud_misc": 12},
+            #     "ft.onto.base_ontology.Document": {"document_class": 4,
+            #               "sentiment": 5, "classifications": 6},
+            #     "ft.onto.base_ontology.Sentence": {"speaker": 4,
+            #               "part_id": 5, "sentiment": 6, "classification": 7,
+            #               "classifications": 8},
+            # }
+        """
+        # TODO: implement get_type_attributes() (Issue #570)
+        self._type_attributes: dict = {}
+
+        """
+        An array that stores a list of elements for each type.
+        It is an underlying storage structure for all the entries added by
+        users in this DataStore.
+
+            Example:
+            self.elements = [
+                Token SortedList(), Document SortedList(),
+                            Sentence SortedList()]
+        """
+        self.elements: List = []
+
+        """
+        The `_type_dict` dictionary stores type id, which is the index of
+        sortedlists in `self.elements`.
+        It should be created only once no matter how many data store objects
+        are initialized.
+
+        Example:
+
+        .. code-block:: python
+
+            self._type_dict: dict = get_type_id()
+
+            # self._type_dict is:
+            # {
+            #     "ft.onto.base_ontology.Token": 0,
+            #     "ft.onto.base_ontology.Document": 1,
+            #     "ft.onto.base_ontology.Sentence": 2,
+            # }
+        """
+        # TODO: implement get_type_id() (Issue #need creation)
+        self._type_dict: dict = {}
+
+        """
+        A dictionary that keeps record of all entrys with their tid.
+        It is a key-value map of {tid: entry data in list format}.
+
+        e.g., {1423543453: [begin, end, tid, entry_type, attr_1, ..., attr_n]}
+        """
+        self.entry_dict: dict = {}
+
+    def _new_entry(self, type: str, begin, end):
+        """
+        generate a new entry with default fields. called by add_entry_raw()
+        to create a new entry with `type`, `begin`, and `end`.
+
+        Args:
+            type (str): type of this entry
+        Returns:
+            A list representing a new entry
+        """
+
+        tid: int = uuid.uuid4().int
+        entry = [begin, end, tid, type]
```

Comment:
Notice here you assume `begin` and `end`, this only works for the `Annotation` type, not for others such as `Link`. Thus this function currently can not create all types of entries. There are two solutions: 1. change this to _new_annotation but then you need functions to create other types 2. make this function generic and can handle all types of entries.